ec3730/Models/DataUsageModel.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  DataUsageModel.swift
+//  ec3730
+//
+//  Created by Ahmad Azam on 21/08/2022.
+//  Copyright © 2022 Zachary Gorak. All rights reserved.
+//
+
+import Foundation
+
+extension SystemDataUsage {
+    static var wifiCompelete: String {
+        let usage = dataUsage.wifiSent + dataUsage.wifiReceived
+        return humanReadableByteCount(bytes: usage)
+    }
+
+    static var wwanCompelete: String {
+        let usage = dataUsage.wirelessWanDataSent + dataUsage.wirelessWanDataReceived
+        return humanReadableByteCount(bytes: usage)
+    }
+
+    static var dataUsage: DataUsageInfo {
+        SystemDataUsage.getDataUsage()
+    }
+
+    static func humanReadableByteCount(bytes: UInt64) -> String {
+        if bytes < 1000 { return "\(bytes) Bytes" }
+        let exp = Int(log2(Double(bytes)) / log2(1000.0))
+        let unit = ["KB", "MB", "GB", "TB", "PB", "EB"][exp - 1]
+        let number = Double(bytes) / pow(1000, Double(exp))
+        if exp <= 1 || number >= 100 {
+            return String(format: "%.0f %@", number, unit)
+        } else {
+            // Can change the decimal percision
+            return String(format: "%.1f %@", number, unit)
+                .replacingOccurrences(of: ".0", with: "")
+        }
+    }
+}
+
+class SystemDataUsage {
+    private static let wwanInterfacePrefix = "pdp_ip"
```

Comment:
Should probably be added to interface tab since this seems so heavily reliant on the interface. Thoughts?

Reply:
I think you're right. So the total in device tab should say wi-fi / cellular or anything else?

Reply:
Could have total, and send/receive.